java/code/src/com/redhat/rhn/frontend/action/common/DownloadFile.java

```diff
@@ -619,7 +619,18 @@ private StreamInfo getStreamInfoKickstart(ActionMapping mapping, ActionForm form
             // my $dp = File::Spec->catfile($kickstart_mount, $tree->base_path, $path);
 
             if (child == null) {
-                diskPath = kickstartMount + "/" + tree.getBasePath() + path;
+                if (path.contains("repodata/") && tree.getKernelOptions().contains("useonlinerepo")) {
+                    String[] split = StringUtils.split(path, '/');
+                    if (split[0].equals("repodata")) {
```

Comment:
These are quite explicit conditions, I'd suggest adding some logging here, which could simplify debugging (at least the final `diskPath`, if it's not logged by something else). That could simplify debugging, in case of problems.

Reply:
diskPath is logged in line 658